### PR TITLE
Place block

### PR DESCRIPTION
**Pull request by caspervdw**
This adds a RasterBlock that can place one raster at given coordinate. For example, you have a 'tree' raster and you want to place it at multiple locations. What you would do is:

```
tree = RasterFileSource("tree.tiff")
forest = Place(tree, "EPSG:3857", [<center of the tree in tree.tiff], [<target coord 1>, <target coord 2>, ...])
```

There are some nice subtleties about reprojections in here. You potentially have 3 different ones:
 1. the native projection of tree.tiff
 2. the projection in which the place is done
 3. the projection of the data request

What I do is:
 - reproject the "anchor" and "coordinates" from the place-projection (2) to the requested projection (3)
 - shift the requested bounding box according to the (anchor - coordinate) differences, all in the requested projection (3)
 - rasterstore takes care about reprojecting from the native one to the requested one (1 -> 3)

The result is that the reprojection takes place at the original coordinate. This might give some funny results (trees may inflate or deflate if you shift them a long distance)

### CODE REVIEW

**Comment by caspervdw** [comment]
https://nelen-schuurmans.atlassian.net/browse/BACK-643

File: dask_geomodeling/raster/spatial.py [hunk 1]
```diff
@@ -429,3 +435,172 @@ def get_sources_and_requests(self, **request):
...
+                    x2 - self.anchor[0] + _x,
+                    y2 - self.anchor[1] + _y,
+                ]
+            )
+
+        # join the extents
+        x1 = min([e[0] for e in extents])
+        y1 = min([e[1] for e in extents])
+        x2 = max([e[2] for e in extents])
+        y2 = max([e[3] for e in extents])
```
Comment (arjanverkerk) [inline dask_geomodeling/raster/spatial.py]: I couldn't resist. What do you think?

```suggestion
         _x1, _x2, _y1, _y2 = store_geometry.GetEnvelope()
         p, q = self.anchor
         P, Q = zip(*self.coordinates)
         x1, x2 = _x1 + min(P) - p, _x2 + max(P) - p
         y1, y2 = _y1 + min(Q) - q, _y2 + max(Q) - q
```

File: dask_geomodeling/raster/spatial.py [hunk 2]
```diff
@@ -429,3 +435,172 @@ def get_sources_and_requests(self, **request):
...
+
+    @staticmethod
+    def process(process_kwargs, *multi):
+        if process_kwargs["mode"] != "vals":
+            return multi[0]
+
+        values = None
+        no_data_value = None
+        for data in multi:
+            if data is None:
```
Comment (arjanverkerk) [inline dask_geomodeling/raster/spatial.py]: This is not tested.

File: dask_geomodeling/raster/spatial.py [hunk 3]
```diff
@@ -429,3 +435,172 @@ def get_sources_and_requests(self, **request):
...
+            if values is None:
+                values = data["values"].copy()
+                no_data_value = data["no_data_value"]
+            else:
+                # index is where the source has data
+                index = get_index(data["values"], data["no_data_value"])
+                values[index] = data["values"][index]
+
+        if values is None:
+            return
```
Comment (arjanverkerk) [inline dask_geomodeling/raster/spatial.py]: Neither is this.

**Comment by caspervdw** [comment]
@arjanverkerk I added a "warp" pathway that is activated when zoomed out enough. A lot of messing around with indices. Tests are pretty thorough so I think it is done now.

**Comment by caspervdw** [comment]
@arjanverkerk  Any chance of a review before the sprint ends?

**Comment by arjanverkerk** [comment]
@caspervdw Not before sprint ends, I guess. I was halfway yesterday, but currently i'm partially looking after the kids and only after lunch i'll be able to work distraction free.

**Comment by caspervdw** [comment]
No problem! I imagine you have a hard time working with all the kids at home

File: dask_geomodeling/raster/spatial.py [hunk 4]
```diff
@@ -429,3 +435,292 @@ def get_sources_and_requests(self, **request):
...
+        full_width = np.ceil((xmax - xmin) / size_x)
+        if full_height * full_width <= request["width"] * request["height"]:
+            _request = request.copy()
+            _request["width"] = int(full_width)
+            _request["height"] = int(full_height)
+            _request["bbox"] = (
+                xmin,
+                ymin,
+                xmin + full_width * size_x,
+                ymin + full_height * size_y,
```
[the remark this answers is not in the document]
Comment (caspervdw) [reply dask_geomodeling/raster/spatial.py]: I also have a hard time of predicting this. I'll set up a test env to actually see the trees.

[the remark this answers is not in the document]
Comment (caspervdw) [reply dask_geomodeling/raster/spatial.py]: Tested, it looks as if it works :)

**Pull request merged**